components/Materiallist.js

```diff
@@ -0,0 +1,229 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { v4 as uuidv4 } from "uuid";
+import SwipeToDeleteItem from "./SwipeToDelete";
+
+export default function Materiallist({
+  addNewDimension,
+  projectId,
+  currentEntry,
+  handleDimensionDelete,
+}) {
+  const [showAddNewDimensions, setShowAddNewDimensions] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    const newDimension = {
+      name: data.name,
+      material: data.material,
+      width: data.width,
+      height: data.height,
+      depth: data.depth,
+      numberOfPieces: data.pieces,
+      unit: data.unit,
+      id: uuidv4(),
+    };
+
+    addNewDimension(projectId, newDimension);
+    event.target.reset();
+  }
+
+  return (
+    <>
+      <Heading>Materiallist</Heading>
+      <ButtonWrapper>
+        <AddNewButton
+          onClick={() => setShowAddNewDimensions(!showAddNewDimensions)}
+        >
+          {showAddNewDimensions ? "click to fold" : "add new dimensions"}
+        </AddNewButton>
+      </ButtonWrapper>
+      {showAddNewDimensions ? (
+        <Form onSubmit={handleSubmit}>
+          <label htmlFor="name">name:</label>
+          <input
+            type="text"
+            id="name"
+            name="name"
+            pattern="^[a-zA-Z0-9äüöÄÜÖ][a-zA-Z0-9-_ äüöÄÜÖß.`'´]{1,}"
+            maxLength={15}
+            placeholder="15 characters allowed"
+            required
+          />
+          <label htmlFor="material">material:</label>
+          <input
+            type="text"
+            id="material"
+            name="material"
+            pattern="^[a-zA-Z0-9äüöÄÜÖ][a-zA-Z0-9-_ äüöÄÜÖß.`'´]{1,}"
+            maxLength={15}
+            placeholder="15 characters allowed"
+          />
+          <label htmlFor="width">width:</label>
+          <input
+            type="text"
+            id="width"
+            name="width"
+            pattern="^[0-9][0-9.,]{1,}"
+            maxLength={5}
+            placeholder="5 characters allowed"
+          />
+          <label htmlFor="height">height:</label>
+          <input
+            type="text"
+            id="height"
+            name="height"
+            pattern="^[0-9][0-9.,]{1,}"
+            maxLength={5}
+            placeholder="5 characters allowed"
+          />
+          <label htmlFor="depth">depth:</label>
+          <input
+            type="text"
+            id="depth"
+            name="depth"
+            pattern="^[0-9][0-9.,]{1,}"
+            maxLength={5}
+            placeholder="5 characters allowed"
+          />
+          <label htmlFor="pieces">number of pieces:</label>
+          <input
+            type="text"
```

Comment:
```suggestion
            type="number"
```

Reply:
damn you right, thats also possible 😅

Reply:
okay i tried, but i couldtn find a way to set maxLength with type="number" so i changed back to type="text". I need that bc i dont want to have endless numbers in the materiallist